doc/reference/common-weaknesses/confused-instance.rst

```diff
@@ -0,0 +1,19 @@
+Separate contract instances may be confused
+===========================================
+
+When a number of on-chain contract instances are started by creating a number of UTXOs addressed to the corresponding script. If there is no unique parameter in the script, then different off-chain instances may address their UTXOs to the same script address, which can lead to confusion and incorrect behavior. (This happens, for example, when the state machine library is used without ‘thread tokens’).
+
+Risks
+~~~~~
+
+Multiple contract instances could get their state confused leading to loss of funds, permanently locked funds or incorrect transfers.
+
+Solutions
+~~~~~~~~~
+
+A possible Solution is to test multiple instances of the same contract against the same emulated blockchain at the same time, interleaving their actions; each instance should separately behave according to the contract model.
+
+Source
```

Comment:
We will probably put a single citation on the first page of this whole section, "Common weaknesses," instead of at the end of each section.

Reply:
Thanks for the suggestion, I was wondering if we should do something like that.

Reply:
Thanks for the suggestion, I was wondering if we should do something like that.